tests/generator/CodeBlockTests.cs

```diff
@@ -0,0 +1,195 @@
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace GeneratorTests {
+	public class CodeBlockTests {
+		string PerformWriting (ICodeBlock codeBlock)
+		{
+			using var writer = new StringWriter ();
+			codeBlock.Print (writer);
+			return writer.ToString ();
+		}
+
+		[Test]
+		public void LineBlockTest ()
+		{
+			string expectedText =
+				"{\n" +
+				"    using System;\n" +
+				"    using Network;\n" +
+				"}\n";
+			CodeBlock codeBlock = new (new string [] { "using System;", "using Network;" });
+			string output = PerformWriting (codeBlock);
+			Assert.AreEqual (expectedText, output);
+
+			expectedText =
+				"{\n" +
+				"    using System;\n" +
+				"    using Network;\n" +
+				"    using NUnit;\n" +
+				"}\n";
+			codeBlock.AddLine ("using NUnit;");
+			output = PerformWriting (codeBlock);
+			Assert.AreEqual (expectedText, output);
+		}
+
+		[Test]
+		public void MethodBlockAddLinesTest ()
+		{
+			string inputText1 = "int fooCount = 1;";
+			string inputText2 = "string[] fooNames = new [] {\"foo\"};";
+			string methodName = "public void Foobinate";
+			string [] methodArguments = new [] { "int count", "bool isFoo" };
+			string expectedText =
+				"public void Foobinate(int count, bool isFoo)\n" +
+				"{\n" +
+				"    int fooCount = 1;\n" +
+				"    string[] fooNames = new [] {\"foo\"};\n" +
+				"}\n";
+			MethodBlock methodBlock = new (methodName, methodArguments);
+			methodBlock.AddLine (inputText1);
+			methodBlock.AddLine (inputText2);
```

Comment:
In general how test code is written doesn't matter all that much, except that for these changes the test code is an example of what kind of code we'll have to write in the generator, so the nicer test code we can write, the nicer code we can write in the generator too.

With that in mind: while using variables for text is generally a good idea, in this case it just makes the code much harder to read, because you have to go back and forth between the declaration and the usage to determine what's going on.

If something like this were possible, it would look much nicer (using a collection initializer):

```cs
var methodBlock = new MethodBlock ("public void Foobinate", "int count", "bool isFoo")
{
	"int fooCount = 1;",
	"string[] fooNames = new [] {\"foo\"};",
};
```

Now if you squint a little bit, it's relatively easy to see what will be generated.